### PR TITLE
Add RFC 176 module definitions

### PR DESCRIPTION
**Pull request by vlascik**

### CODE REVIEW

**Comment by dwickern** [comment]
@vlascik Could you keep the same indent level like I did in [my ember-data PR](https://github.com/typed-ember/ember-typings/blob/649f52155e6afffeb5a2d148d4deb35cdfbb8c2c/types/ember/index.d.ts)? That should make it easier for me to merge #1

File: types/ember/index.d.ts [hunk 1]
```diff
@@ -2637,3 +2638,358 @@ declare namespace Ember {
 }
 
 export default Ember;
+}
+
+declare module '@ember/application' {
+    import Ember from 'ember';
+    export default class Application extends Ember.Application { }
```
Comment (dwickern) [inline types/ember/index.d.ts]: Can this be `export default Ember.Application`?

Comment (chriskrycho) [inline types/ember/index.d.ts]: Interesting; this also [fails the lint](https://travis-ci.org/typed-ember/ember-typings/builds/264796866#L558):

> strict-export-declare-modifiers  'export' keyword is redundant here.

**Comment by chriskrycho** [comment]
@vlascik can you add some tests to import from these? I'd just like something in place that way since we're doing some pretty complicated stuff.

**Comment by dwickern** [comment]
The [RFC](https://github.com/emberjs/rfcs/blob/master/text/0176-javascript-module-api.md) has some examples for the test suite

**Comment by dwickern** [comment]
I had a look at your branch. The `export` keyword actually *is* needed in `export const or: typeof Ember.computed.or;`

That tslint rule `strict-export-declare-modifiers` is wrong, I think we should disable it.

**Comment by chriskrycho** [comment]
This isn't argument but clarification: is that rule on in DT? Because if so we're going to have to figure out some way to deal with it. 😕

**Comment by dwickern** [comment]
It's in `ember`'s tslint, so we can disable it. If we can get the tests to pass here, they should pass on DT

**Pull request merged**